packages/features/ee/teams/lib/payments.ts

```diff
@@ -29,6 +29,51 @@ export const checkIfTeamPaymentRequired = async ({ teamId = -1 }) => {
   return { url: `${WEBAPP_URL}/api/teams/${teamId}/upgrade?session_id=${metadata.paymentId}` };
 };
 
+/**
+ * Used to generate a checkout session when trying to create a team
+ */
+export const generateTeamCheckoutSession = async ({
```

Comment:
Created a new function to generate a checkout session for this scenario. I want to keep the previous function that was used `purchaseTeamSubscription` for backwards compatibility for unpublished teams and orgs.